CHANGES.rst

```diff
@@ -7,6 +7,8 @@ Unreleased
 **Changed**
 
 - :meth:`.Redditor.moderated` will now objectify all data returned from the API.
+- The `wiki_edit` endpoint has been changed from `r/{subreddit}/api/wiki/edit/` to
+  `r/{subreddit}/api/wiki/edit`.
 
```

Comment:
Could we add a fixed entry as well? Since it does fix an issue.

Reply:
Something like

- The `wiki_edit` endpoint has been changed from `r/{subreddit}/api/wiki/edit/` to
  `r/{subreddit}/api/wiki/edit`. This allows :meth:`.WikiPage.edit` to set its `content`
  parameter to a string over 500 KiB when creating or editing wiki pages with special
  upload limits, e.g. `usernotes`.

?

Reply:
The latter. Keep the `**Changed**` and add a `**Fixed**`.

Reply:
A few more things:

The current normal wikipage size limit is 512 KiB, not 500 KiB, so this change affects normal wiki pages as well. With `edit`, the API returns a 403 with an error message of `{"reason": "CONTENT_LENGTH_ERROR", "max_length": 524288, "message": "Forbidden"}` when you try to make the `content` of a normal page to be over 512 KiB.

With `edit/`, requests fail when the body of the POST request is over 500 KiB. That number includes the lengths of the keys, & separators and = signs, which is something you might not expect. Since `api_type=json&content=&page=usernotes` is 37 characters, the previous actual maximum length you could set with the API was 512,000 - 37= 519963 bytes, and less if you included a reason. With `edit`, the same kind of limitations apply to the usernotes page: the upper limit is a request body of 1 MiB, not a `content` length of 1 MiB. This is probably a bug. Fortunately it doesn't seem to affect normal wiki pages, which can be set to their maximum content length with a reason of maximum length. 

The maximum allowed length for the `reason` parameter is 256 characters, just as the docs say. However, if you try to set a  reason over 256 chars, you get a 400 response with a message that says `{"reason": "TOO_LONG", "message": "Bad Request", "explanation": "This field must be under %(max_length)s characters"}`. This is also probably a bug. I can share these tests later today.

Reply:
Let me know if there is something else you want to add before I merge.

Reply:
Thanks for taking care of the issue with the actions. The last thing to check was to see if it works with the `previous` parameter, which it does, so it can be merged. I could edit (now or later) the original text of the pull request to make it clear that this change affects normal wiki pages as well.

Reply:
Other things: I can see why a method for *completing* a safe edit was [added](
https://github.com/praw-dev/praw/pull/816) to prawadditions instead of PRAW. It might be worth adding something in the docs about how the `previous` parameter is used: Reddit tries to try to do a [three way merge](https://github.com/reddit-archive/reddit/blob/master/r2/r2/models/wiki.py#L398) via [diff3](https://github.com/reddit-archive/reddit/blob/master/r2/r2/lib/merge.py) with the `content` you pass in, the current revision, and the `previous` revision as the common ancestor. If there is a merge conflict, it prevents you from editing. In practice, if there is a conflict, you'd probably want to resolve it manually, which I guess is why Reddit's response isn't really useful for programmatic resolution. I realize that the archived version of r2 is out of date, so maybe just linking to the Wikipedia page about three way merges would help newcomers or people not familiar with the endpoint. 

Do you know why the json and non-json versions of the /wiki/page endpoint are different? The docs say that if you use both v and v2, it returns a diff of the two pages. But the json version doesn't return a diff. Compare [this](https://www.reddit.com/r/redditdev/wiki/config/stylesheet?v=8deb1630-6e67-11e6-a946-0e02b31a9371&v2=5879bd1c-f66a-11e7-b446-0e04dda67410) and [this](https://www.reddit.com/r/redditdev/wiki/config/stylesheet.json?v=8deb1630-6e67-11e6-a946-0e02b31a9371&v2=5879bd1c-f66a-11e7-b446-0e04dda67410&raw_json=1). The API returns a difflib HTML table when the edit endpoint has a merge conflict, so I don't see why it it couldn't be included here. I know that there are other endpoints, like /about/traffic, which are also different in this way, but I'm not sure if this one is intentionally different.

Finally, (unless I've missed something), why doesn't PRAW have anything that uses the wiki/hide and wiki/revert endpoints? Hiding a revision isn't the same thing as delisting a page.

Reply:
Sounds good, I'll get this merged in.

Regarding the second comment, that's not something I am too sure on, @bboe might have answers for you on those points.

Reply:
> Do you know why the json and non-json versions of the /wiki/page endpoint are different? The docs say that if you use both v and v2, it returns a diff of the two pages. But the json version doesn't return a diff. Compare this and this. The API returns a difflib HTML table when the edit endpoint has a merge conflict, so I don't see why it it couldn't be included here.

It's not uncommon for the html and json responses to differ. This issue would be something worth reporting to https://www.reddit.com/r/bugs/. Back when Reddit was open sourced you could make a PR to add that diff attribute to the JSON yourself, alas those days are no more.

> Finally, (unless I've missed something), why doesn't PRAW have anything that uses the wiki/hide and wiki/revert endpoints? Hiding a revision isn't the same thing as delisting a page.

If a feature is missing, that means no one has ever taken the time to implement it. Looks like a perfect opportunity for you to add it to PRAW :)

Thanks for the contribution!